### PR TITLE
promises disallowed

### PR DESCRIPTION
**Pull request by TG1999**
Signed-off-by: tushar goel <tushar.goel.dav@gmail.com>

This patch includes:
- [ ] A relevant test
- [ ] A relevant documentation update

### CODE REVIEW

File: lib/productions/type.js [hunk 1]
```diff
@@ -89,8 +89,9 @@ function union_type(tokeniser, type) {
   const ret = new Type({ source: tokeniser.source, tokens });
   ret.type = type || null;
   while (true) {
+    if(tokeniser.probe("any")) tokeniser.error("Type `any` cannot be included in a union type");
```
Comment (saschanaz) [inline lib/productions/type.js]: I'm sorry to say this, but I found a bug which blocks this change (#436), which is why `union-any.txt` is now gone empty. Could you revert this change to match what you did in #431? This includes the Promise line below, because this way we can't prevent `[ExtAttr] Promise` being included in a union. Sorry to confuse you 😢

**Comment by TG1999** [comment]
##[error]  94:105  error  Trailing spaces not allowed  no-trailing-spaces @saschanaz this is the error

File: lib/productions/type.js [hunk 2]
```diff
@@ -91,6 +91,7 @@ function union_type(tokeniser, type) {
   while (true) {
     const typ = type_with_extended_attributes(tokeniser) || tokeniser.error("No type after open parenthesis or 'or' in union type");
     if (typ.idlType === "any") tokeniser.error("Type `any` cannot be included in a union type");
+    if (typ.generic === "Promise") tokeniser.error("Type `Promise` cannot be included in a union type");
```
Comment (saschanaz) [inline lib/productions/type.js]:
```suggestion
    if (typ.generic === "Promise") tokeniser.error("Type `Promise` cannot be included in a union type");
```

😁

**Comment by TG1999** [comment]
@saschanaz looks good to merge ?

File: test/invalid/idl/union-promise.webidl [hunk 3]
```diff
@@ -0,0 +1 @@
+ typedef (Promise<DOMString> or boolean) Typedef;
```
Comment (saschanaz) [inline test/invalid/idl/union-promise.webidl]:
```suggestion
typedef (Promise<DOMString> or boolean) Typedef;
```

**Comment by saschanaz** [comment]
Could you do `git checkout HEAD~1 dist` once again? We don't want to update them except when we do a release.

**Comment by TG1999** [comment]
I have pasted the same dist files that are in ```gh-pages``` so it should not make any changes

**Comment by saschanaz** [comment]
Yes... but it caused newline change as GitHub always adds a newline to the end of the file.

**Comment by saschanaz** [comment]
I can do it myself if you check ["Allow edits from maintainers"](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) as you already did a great job 👍

**Comment by TG1999** [comment]
I have ran the command but nothing is getting changed, can you please make edits in it :pray:

**Comment by saschanaz** [comment]
Oops, the correct command is `git checkout upstream/gh-pages dist`

File: test/invalid/idl/union-promise.webidl (hunk 3 above)
Comment (saschanaz) [inline test/invalid/idl/union-promise.webidl]:
```suggestion
typedef (Promise<DOMString> or boolean) Typedef;
```

**Comment by TG1999** [comment]
@saschanaz any more issues for me 😊
